### DISCUSSION TITLE
Why there is a sleep function

### DISCUSSION THREAD

**Discussion by Tang115** [CURENT/andes#271] [General]
Dr. Cui,

When I read the source code in Andes, I see there is a sleep function in TDS.run():
https://github.com/cuihantao/andes/blob/8d484f1ddd02abf71788f2a545c7cbaa152eca73/andes/routines/tds.py#L358 

What is the role of this sleep function in TDS. run()? What are we waiting for by setting up this delay?

Best,
Thank you!

**Comment by cuihantao** [comment]
Hi,

The sleep function is to implement a so-called quasi-realtime function. It mimics a real-time simulator when the simulation speed is faster than the wall clock speed.

**Reply by cuihantao** [reply]
From command line, the quasi-realtime can be tested with `andes run case.xlsx -r tds --qrt`.
